### ISSUE TITLE
Unable to find Expected Key - File issue?

### ISSUE THREAD

**Issue by Luzr** [peeley/carpal#1]
Hi,

When I try to webfinger my domain, tailscale OIDC setup, running carpal as a docker container I get the error below. I thought it was the location of the acct: file but I'm unsure as I cant open a bash terminal into the container. 

```console
Attaching to carpal-1
carpal-1  | 2024/04/14 14:24:05 launching carpal server on port 8008...
carpal-1  | 2024/04/14 14:24:10 received request for resource acct:jdoe@domazor.com
carpal-1  | 2024/04/14 14:24:10 unable to unmarshal resource file contents: yaml: line 5: did not find expected key
carpal-1  | 2024/04/14 14:24:10 resource acct:jdoe@domazor.com not found: could not unmarshal file to JRD: yaml: line 5: did not find expected key
```

Here are my config files and their contents.

```console
root@docker:~/webfinger# cat docker-compose.yml 
version: "3"

services:
  carpal:
    image: peeley/carpal:latest
    volumes:
      - ./config:/etc/carpal/
    ports:
      - 8090:8008
```

```console
root@docker:~/webfinger# cat /etc/carpal/config.yml 
# /etc/carpal/config.yml

driver: file
file:
  directory: /etc/carpal/resources/
```

```console
root@docker:~/webfinger# cat /etc/carpal/resources/acct\:jdoe@domazor.com 
# config/resources/acct:jdoe@domazor.com

aliases:
  - "mailto:jdoe@domazor.com"
links:
  - rel: "http://openid.net/specs/connect/1.0/issuer"
    href: ""https://oauth.id.jumpcloud.com/"
root@docker:~/webfinger# 
```

**Comment by peeley** [comment]
Hmm, based on the error message `could not unmarshal file to JRD: yaml: line 5: did not find expected key`, it seems like the file exists in the container but it's erroring when trying to parse the YAML.

Not sure if it's the root of the issue, but it looks like in the `acct` file under the `links` field, the value for `href` has an extra quotation mark (`""https://oauth.id.jumpcloud.com/"`). Maybe try this:

```
# config/resources/acct:jdoe@domazor.com

aliases:
  - "mailto:jdoe@domazor.com"
links:
  - rel: "http://openid.net/specs/connect/1.0/issuer"
    href: "https://oauth.id.jumpcloud.com/"
```

**Comment by Luzr** [comment]
Thanks for the response and double checking. Your correct. It was the extra double quote.

Just to verify, the docker container checks the local path config, like below, for the acct files or does it check /etc/carpal on the hos?

```
|-- config
|   |-- config.yml
|   `-- resources
|       |-- acct:jdoe@domazor.com
`-- docker-compose.yml
```

**Comment by peeley** [comment]
No worries, happy to help!

> Just to verify, the docker container checks the local path config, like below, for the acct files or does it check /etc/carpal on the host?

Kind of; it checks the local path in the sense that if you mount that local directory into the Docker container at `/etc/carpal`, it'll be able to access your local directory through the container filesystem's `/etc/carpal` directory. The Docker container should only have access to the volumes that you mount to it via the `docker-compose.yml` file. The container isn't able to access anything "outside" of the container, so it can't check the `/etc/carpal` directory (or the local directory) on the host unless you give it explicit access to do so.

In short - yes, the example `docker-compose.yml` file only checks the host's local `config` directory and not the `/etc/carpal` directory.
